modules/basics/src/main/java/com/opengamma/strata/basics/index/FxIndex.java

```diff
@@ -47,7 +49,47 @@
   @FromString
   public static FxIndex of(String uniqueName) {
     ArgChecker.notNull(uniqueName, "uniqueName");
-    return extendedEnum().lookup(uniqueName);
+    try {
+      return extendedEnum().lookup(uniqueName);
+    } catch (IllegalArgumentException ex) {
+      CurrencyPair currencyPair = CurrencyPair.parse(uniqueName);
```

Comment:
Needs to be a try-catch around this line. The idea is that the error message should refer to "index not found", not "unable to parse currency pair".